app/js/lib/foxbox/foxbox.js

```diff
@@ -180,32 +149,29 @@ export default class Foxbox extends Service {
         // we have no record of previous registrations, we schedule
         // a retry.
         if (!this[p.boxes].length &&
-            !this[p.settings]._localOrigin &&
-            !this[p.settings]._tunnelOrigin) {
+            !this[p.settings].localOrigin &&
+            !this[p.settings].tunnelOrigin) {
           throw new Error('Registration service did not return any boxes.');
         }
 
         if (!this[p.settings].configured) {
           this.selectBox();
         }
-
-        this._dispatchEvent('box-online', true);
```

Comment:
So, the easiest way (to not make this PR even more bigger :)) is to introduce ```discovery``` event and call it every time registration server returns a valid response (if we override ```this[p.boxes]``` I'll fire event). Later on we can improve logic to check if box list has actually changed.